docs/webiny-apps/page-builder/development/creating-iframe-element-plugin.md

```diff
@@ -0,0 +1,203 @@
+import IFrameEmbed from "./iFrameEmbed";
+
+const PreviewBox = styled("div")({
+    textAlign: "center",
+    height: 50,
+    svg: {
+        height: 50,
+        width: 50
+    }
+});
+
+export default () => {
+    return {
+            name: "pb-editor-page-element-iframe",
+            type: "pb-editor-page-element",
+            elementType: "iframe",
+            toolbar: {
+                title: "iFrame",
+                group: "pb-editor-element-group-media",
+                preview() {
+                    return (
+                        <PreviewBox>
+                            <IFrameIcon />
+                        </PreviewBox>
+                    );
+                }
+            },
+            settings: ["pb-editor-page-element-settings-delete"],
+            onCreate: "open-settings",
+            create(options) {
+            // Create function it's here to create the initial data for the page element, which then is utilized in theIFrameEmbed, in settings dialog, etc.
+                return {
+                    type: "iframe",
+                    elements: [],
+                    data: {
+                        iframe: {
+                            // The URL property will be populated when user enters the URL in the settings dialog
+                            url: "",
+                            height: 370
+                        },
+                        settings: {
+                            horizontalAlign: "center",
+                            margin: {
+                                desktop: { all: 0 },
+                                mobile: { all: 0 }
+                            },
+                            padding: {
+                                desktop: { all: 0 },
+                                mobile: { all: 0 }
+                            }
+                        }
+                    },
+                    ...options
+                };
+            },
+            render(props) {
+                return <IFrameEmbed {...props} />;
+            },
+            renderElementPreview({ width, height }) {
+                return <img style={{ width, height }} alt={"iFrame"} />;
+            }
+        } as PbEditorPageElementPlugin,
+    // Settings dialog comes here
+```
+
+Since we are adding a new element in the editor, the type of plugin will be `pb-editor-page-element`. This plugin consists of different properties, one of them is the `toolbar` property that helps us put our element into the "Media" group, using `pb-editor-element-group-media`, as seen in the image below.
+
+The key properties of the plugin are the `create`, `data`, `settings`, and `render`. They contribute to the state of our plugin. On changing the state, it’s up to the render function to render the component based on the state changes.
+
+![Editor Iframe Element](/img/webiny-apps/page-builder/development/development/plugin-reference/editor/iframe/editor-iframe-plugin.png)
+
+### Render Plugin
```

Comment:
Can you change the color of the arrow? For example red?

I actually didn't see it instantly, because it has a very similar color to the surrounding UI. It's not popping up, other users might miss it too.